app/routes/task.py

```diff
@@ -0,0 +1,143 @@
+        response_str = "Invalid data"
+        abort(make_response({"details":response_str}, 400))
+    if "completed_at" not in request_body:
+        request_body["is_complete"] = False
+
+    new_task = Task.from_dict(request_body)
+    db.session.add(new_task)
+    db.session.commit()
+    task_dict = new_task.make_dict()
+    response = {"task": task_dict}
+    return make_response(response, 201)
+    
+@task_bp.route("/<task_id>", methods = ["PUT", "PATCH"])
+def update_task(task_id):
+    task = validate_task(task_id)
+    request_body = request.get_json()
+    task = update_given_values(task, request_body)
+    db.session.commit()
+    response = {"task": task.make_dict()}  #refactor this line and line 37 above to helper function? or method on class?
+    return make_response(response, 200)
+
+#can the following be combined with the route above?
+@task_bp.route("/<task_id>/<complete_tag>", methods = ["PUT", "PATCH"])
+def update_task_completion(task_id, complete_tag):
+    task = validate_task(task_id)
+    if complete_tag == "mark_complete":
+        task.completed_at = date.today().strftime("%B %d, %Y")
+        task.is_complete = True  #not sure if this line is redundant
+
+        #post a message to slack to say the task is complete.
+        #----------Make the following into a helper function later-----#
+        path = "https://slack.com/api/chat.postMessage"
+        query_params = {
+            "channel" : "task-notifications",
+            "text": "Someone just completed the task " + task.title
+        }
+        headers = {
+            "Authorization" : "Bearer " + os.environ.get("SLACK_API_KEY")
+        }
+
+        response = requests.post(path, params = query_params, headers = headers)
+        #----------end helper function--------------#
+
+    elif complete_tag == "mark_incomplete":
+        task.completed_at = None
+        task.is_complete = False #not sure if this line is redundant
+    db.session.commit()
+    response = {"task": task.make_dict()}
+    return make_response(response, 200)
+
+
+@task_bp.route("/<task_id>", methods = ["DELETE"])
+def delete_task(task_id):
+    task = validate_task(task_id)
+    db.session.delete(task)
+    db.session.commit()
+    response_body = {"details": f'Task {task_id} "{task.title}" successfully deleted'}
+    return make_response(response_body, 200)
+
+#ideally, combine this with validate goal, passing in the class as well. 
+def validate_task(task_id):
+    try:
+        task_id = int(task_id)
+    except ValueError:
+        response_str = f"Task {task_id} must be an integer"
+        abort(make_response({"message": response_str}, 400))
+    task = Task.query.get(task_id)
+    if not task:
+        response_str = f"Task {task_id} not found"
+        abort(make_response({"message": response_str}, 404))
+    return task
+
+#can I make this a method for Tasks?
+def update_given_values(task, request_body):
```

Comment:
You can indeed make this a method in the tasks class.  It would make a great helper method.